src/Features/Core/Portable/StackTraceExplorer/StackTraceExplorerService.cs

```diff
@@ -105,7 +105,11 @@ private static ImmutableArray<TextDocument> GetFileMatches(Solution solution, St
 
             foreach (var document in allDocuments)
             {
-                if (string.Equals(document.Name, documentName, StringComparison.OrdinalIgnoreCase))
+                if (document.FilePath?.EndsWith(documentName, StringComparison.OrdinalIgnoreCase) is true)
+                {
+                    potentialMatches.Add(document);
+                }
+                else if (document.Name.Equals(documentName, StringComparison.OrdinalIgnoreCase))
```

Comment:
The `documentName` shouldn't have a path separator so could we just combine the two checks like so?
```suggestion
                if (document.Name?.EndsWith(documentName, StringComparison.OrdinalIgnoreCase) is true)
```

Reply:
Idk if it's just a test setup issue but I encountered times where the name was the full path

Reply:
> I encountered times where the name was the full path

Does that validate my suggestion to only check `document.Name.EndsWith`?

Reply:
Or are you saying `var documentName = Path.GetFileName(fileName);` returns the full path in some instances? (Maybe windows line endings on a unix-like platform)

Reply:
I'm saying sometimes Name is a full path and sometimes it's not. On phone so may be missing something but this check at least seems slightly closer to intent